### PR TITLE
Fix task creation with `gt_pool` validation and cloud storage data

### PR DESCRIPTION
**Pull request by Marishka17**
### How has this been tested?

Added corresponding REST API test
### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have created a changelog fragment 
~~- [ ] I have updated the documentation accordingly~~
- [x] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
~~- [ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))~~

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

## Summary by CodeRabbit

- **New Features**
	- Enhanced task creation process with improved validation for segment sizes and parameters.
	- New method for reordering images in the manifest based on user-defined lists.

- **Bug Fixes**
	- Improved handling of manifest files and error handling for invalid configurations during task creation.

- **Tests**
	- Expanded test suite for task management, focusing on validation frames and cloud data integration.

- **Chores**
	- Updates to function signatures and type annotations for better clarity and maintainability.

### CODE REVIEW

File: tests/python/shared/utils/helpers.py [hunk 1]
```diff
@@ -35,7 +39,9 @@ def generate_image_files(
     for i in range(count):
         prefix = prefixes[i] if prefixes else ""
         filename = f"{prefix}{i}.jpeg" if not filenames else filenames[i]
-        image = generate_image_file(filename, color=(i, i, i))
+        image = generate_image_file(
+            filename, color=(i, i, i), **({"size": sizes[i]}) if sizes else {}
```
[the remark this answers is not in the document]
Comment (Marishka17) [reply tests/python/shared/utils/helpers.py:43]: There is no need to define one more place with the default `size` value.

File: tests/python/rest_api/test_tasks.py [hunk 2]
```diff
@@ -2550,6 +2554,85 @@ def test_can_create_task_with_gt_job_from_video(
...
+        validation_frames = ["test/sub_0/img_1.jpeg"]
+
+        (task_id, _) = self._create_task_with_cloud_data(
+            request,
+            cloud_storage,
+            use_manifest=False,
+            server_files=server_files,
+            sorting_method=models.SortingMethod(
+                "random"
+            ),  # only random sorting can be used with gt_pool
```
[the remark this answers is not in the document]
Comment (Marishka17) [reply tests/python/rest_api/test_tasks.py:2585]: no need

File: tests/python/rest_api/test_tasks.py [hunk 3]
```diff
@@ -2550,6 +2554,85 @@
...
+                    chunk_images = {
+                        int(os.path.splitext(name)[0]): np.array(
+                            Image.open(io.BytesIO(chunk_archive.read(name)))
+                        )
+                        for name in chunk_archive.namelist()
+                    }
+                    chunk_images = dict(sorted(chunk_images.items(), key=lambda e: e[0]))
+
+                    for img, img_meta in zip(chunk_images.values(), job_meta.frames):
+                        assert (img.shape[0], img.shape[1]) == (img_meta.height, img_meta.width)
```
[the remark this answers is not in the document]
Comment (Marishka17) [reply tests/python/rest_api/test_tasks.py:2634]: no need

**Review by Bobronium — APPROVED** [review]

**Review by bsekachev — APPROVED** [review]
LGTM

**Comment by Marishka17** [comment]
Failed checks (`License Compliance`, `Security Analysis`) are not related to the PR, I'll merge them.

**Pull request merged**